blog/utils/handlePosts.ts

```diff
@@ -43,10 +43,21 @@ export const sortPosts = (blogPosts: BlogPost[], sortBy: SortBy) => {
  * @param posts Posts to be handled
  * @param slug Category Slug to be filter
  */
-export const filterPostsByCategory = (posts: BlogPost[], slug?: string) =>
+export const filterPostsByCategory = (
+  posts: BlogPost[],
+  slug?: string,
+  term?: string,
+) => (
   slug
-    ? posts.filter(({ categories }) => categories.find((c) => c.slug === slug))
-    : posts;
+    ? posts.filter(({ categories }) => categories.some((c) => c.slug === slug))
+    : term
+    ? posts.filter(({ content, excerpt, title }) =>
+      [content, excerpt, title].some((field) =>
+        field.toLowerCase()?.includes(term.toLowerCase())
+      )
+    )
+    : posts
+);
```

Comment:
change to early return

Reply:
maybe this function has to be the name changed or spllited in separated functions